### PR TITLE
REAL-2514 implement mock contract event testing for atlas flow

### PR DESCRIPTION
**Pull request by gheorghestrimtu**
To improve the testing process of the Atlas flow, which involves reading events from the blockchain and saving them in the database, we need to generate mock contracts that emit the same events as our actual contracts without executing the original contract logic. The mock contracts should be generated automatically based on our existing Solidity contracts.

This task involves creating a script to generate mock contracts, integrating the script into our existing contract workflow, updating the GitHub Actions pipeline.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,6 +5,7 @@ go 1.20
 require (
 	github.com/Depado/ginprom v1.7.4
 	github.com/Masterminds/semver/v3 v3.2.0
+	github.com/Masterminds/sprig v2.22.0+incompatible
```
Comment (jmank88) [inline go.mod:8]: Is there a newer version of this module that uses `github.com/Masterminds/semver/v3` which we already import?

Comment (gheorghestrimtu) [reply go.mod:9]: Getting this if I try to update to latest version
```go: errors parsing go.mod:
/Users/gheorghestrimtu/Documents/chainlink/git/chainlink/go.mod:9:2: require github.com/Masterminds/sprig: version "v3.2.3" invalid: should be v0 or v1, not v3
```

Comment (jmank88) [reply go.mod:9]: I think you have to include the `/v3` suffix too. I guess they didn't follow that pattern with v2?
```suggestion
	github.com/Masterminds/sprig/v3 v3.2.3
```

Comment (gheorghestrimtu) [reply go.mod]: worked

File: GNUmakefile [hunk 2]
```diff
@@ -67,7 +67,6 @@ abigen: ## Build & install abigen.
 
 .PHONY: go-solidity-wrappers
 go-solidity-wrappers: pnpmdep abigen ## Recompiles solidity contracts and their go wrappers.
-	./contracts/scripts/native_solc_compile_all
```
Comment (jmank88) [inline GNUmakefile:70]: This seems important, no?

Comment (gheorghestrimtu) [reply GNUmakefile:70]: this script is run first thing here 
`https://github.com/smartcontractkit/chainlink/blob/develop/core/gethwrappers/go_generate.go#L6`
this line was just a duplicate

Comment (jmank88) [reply GNUmakefile:70]: Hmmm are you sure? I see `./generation/compile_contracts.sh` at that link, and the failing solidity tests run `make go-solidity-wrappers` directly: https://github.com/smartcontractkit/chainlink/blob/f2b10ae4d6e91159692e2a0e3757a2f694dc2d9c/.github/workflows/solidity.yml#L218
https://github.com/smartcontractkit/chainlink/actions/runs/4495358702/jobs/7908898697#step:7:1

Comment (gheorghestrimtu) [reply GNUmakefile:70]: https://github.com/smartcontractkit/chainlink/blob/develop/core/gethwrappers/generation/compile_contracts.sh#L6

File: core/gethwrappers/generation/compile_event_mock_contract.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
```
Comment (cll-gg) [inline core/gethwrappers/generation/compile_event_mock_contract.sh:2]: You could add `set -euo pipefail` (see http://redsymbol.net/articles/unofficial-bash-strict-mode/)

Comment (gheorghestrimtu) [reply core/gethwrappers/generation/compile_event_mock_contract.sh:2]: done

**Pull request merged**